src/viewport/observability.py

```
@@ -0,0 +1,303 @@
+        async_engine = get_engine()
+        sync_engine = getattr(async_engine, "sync_engine", None)
+        if sync_engine is not None:
+            SQLAlchemyInstrumentor().instrument(engine=sync_engine)
+        SQLAlchemyInstrumentor().instrument(engine=get_sync_engine())
+        _DB_INSTRUMENTED = True
+    except Exception:
+        logger.debug("OpenTelemetry SQLAlchemy instrumentation unavailable", exc_info=True)
+
+
+def _endpoint_is_insecure(endpoint: str) -> bool:
+    return endpoint.startswith("http://") or endpoint.startswith("localhost") or endpoint.startswith("127.0.0.1")
```

Ruff (0.15.12)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)